src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/UDPCommunicator.java

```diff
@@ -0,0 +1,493 @@
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
+		} else {
+			this.timeout = timeout;
+		}
+	}
+
+	/**
+	 * This method returns the device UPD buffer length
+	 *
+	 * @return int This returns the current UDP buffer length.
+	 */
+	public int getBufferLength() {
+		return this.bufferLength;
+	}
+
+	/**
+	 * This method is used set the device UDP buffer length
+	 *
+	 * @param bufferLength This is the UDP buffer length to set
+	 */
+	public void setBufferLength(int bufferLength) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
+		} else {
+			this.bufferLength = bufferLength;
+		}
+	}
+
+	/**
+	 * This method returns the device UPD port
+	 *
+	 * @return int This returns the current UDP port.
+	 */
+	public int getPort() {
+		return this.port;
+	}
+
+	/**
+	 * This method is used set the device UDP port
+	 *
+	 * @param port This is the UDP port to set
+	 */
+	public void setPort(int port) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
+		} else {
+			this.port = port;
+		}
+	}
+
+	/**
+	 * This method returns the list of command error
+	 *
+	 * @return List<String> This returns the current list of command error
+	 */
+	public List<String> getCommandErrorList() {
+		return this.commandErrorList;
+	}
+
+	/**
+	 * This method is used set the list of command error
+	 *
+	 * @param commandErrorList This is the list of command error to set
+	 */
+	protected void setCommandErrorList(List<String> commandErrorList) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
+		} else {
+			this.commandErrorList = commandErrorList;
+		}
+	}
+
+	/**
+	 * This method returns the list of command success
+	 *
+	 * @return List<String> This returns the current list of command success
+	 */
+	public List<String> getCommandSuccessList() {
+		return this.commandSuccessList;
+	}
+
+	/**
+	 * This method is used set the list of command success
+	 *
+	 * @param commandSuccessList This is the list of command success to set
+	 */
+	protected void setCommandSuccessList(List<String> commandSuccessList) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
+		} else {
+			this.commandSuccessList = commandSuccessList;
+		}
+	}
+
+	/**
+	 * This method is used to create a connection
+	 */
+	@Override
+	public void connect() {
+		if (!this.isInitialized()) {
+			throw new IllegalStateException("UDPCommunicator cannot be used before init() is called");
+		} else {
+			if (this.logger.isTraceEnabled()) {
+				this.logger.trace("Connecting to: " + this.host + " port: " + this.port);
+			}
+
+			Lock writeLock = this.lock.writeLock();
+			writeLock.lock();
+
+			try {
+				if (!this.isChannelConnected()) {
```

Comment:
Which exception can be thrown in this try block? As I see, no exception can be thrown here. The `createChannel()` is also catch all exceptions and only logging it in debug mode. Please correct me if I'm wrong.

Reply:
ok, I'll update it